benchmark_runner/common/ocp_resources/create_odf.py

```
@@ -33,6 +34,19 @@ def create_odf(self, upgrade_version: str):
+                idms_path = os.path.join(self.__path, 'idms.yaml')
+                logger.info(f'Extracting idms.yaml from catalog image: {self.__odf_catalog_image}')
+                self.__oc.run(cmd=f"cd {self.__path} && oc image extract '{self.__odf_catalog_image}' --file='/idms.yaml' --confirm")
+                if os.path.exists(idms_path) and open(idms_path).read().strip():
+                    logger.info('Applying IDMS for catalog image mirrors')
+                    self.__oc.run(cmd=f'oc apply -f {idms_path}')
+                    logger.info('Waiting for MCP rollout after IDMS apply')
+                    self.wait_for_ocp_resource_create(operator='odf',
+                                                      verify_cmd="oc get mcp -o jsonpath='{range .items[*]}{.status.conditions[?(@.type==\"Updated\")].status}{\"\\n\"}{end}' | grep -c True || true",
+                                                      status=str(len(['master', 'worker'])))
+                else:
+                    logger.info(f'No idms.yaml found in catalog image, skipping IDMS apply')
```

Ruff (0.16.1)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)